### PR TITLE
Dt rk #164080608 fetch accessorial errors

### PR DESCRIPTION
**Pull request by Ryan-Koch**
## Description
These are the things going on in this one:

* Creates the start of a service object for ShipmentLineItems and moves the Index/Get and Recalculate functionality over to that model. 
* Breaks out Index/Get and Recalculate into two separate handlers.
* Implements mockery for handler tests and sets up service tests 
* Creates a noticeable error alert when Shipment Line Items fail to recalculate

## Setup

Add any steps or code to run in this section to help others prepare to run your code:

```sh
make db_dev_e2e_populate
make server_run
make client_run
```

## Code Review Verification Steps

* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/tree/master/docs/backend.md#logging)
* [ ] The requirements listed in
 [Querying the Database Safely](https://github.com/transcom/mymove/tree/master/docs/backend.md#querying-the-database-safely)
 have been satisfied.
* Any new migrations/schema changes:
  * [ ] Follow our guidelines for zero-downtime deploys (see [Zero-Downtime Deploys](https://github.com/transcom/mymove/tree/master/docs/database.md#zero-downtime-migrations))
  * [ ] Have been communicated to #dp3-engineering
  * [ ] Secure migrations have been tested using `scripts/run-prod-migrations`
* [ ] There are no aXe warnings for UI.
* [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
* [ ] User facing changes have been reviewed by design.
* [ ] Request review from a member of a different team.
* [ ] Have the Pivotal acceptance criteria been met for this change?

## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/164080608) for this change

## Screenshots
image

### CODE REVIEW

File: pkg/services/shipment_line_item/get_shipment_line_items.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+// IndexStorageInTransits returns a collection of Storage In Transits that are associated with a specific shipmentID
+func (i *getShipmentLineItems) GetShipmentLineItemsByShipmentID(shipmentID uuid.UUID, session *auth.Session) ([]models.ShipmentLineItem, error) {
+
+	if session.IsTspUser() {
+		// Check that the TSP user can access the shipment
+		_, _, err := models.FetchShipmentForVerifiedTSPUser(i.db, session.TspUserID, shipmentID)
+		if err != nil {
+			return nil, err
```
Comment (lynzt) [inline pkg/services/shipment_line_item/get_shipment_line_items.go]: do we want to return a reason for the error? 
something like: 
`h.Logger().Error("Reason why here...", zap.Error(err))`

File: pkg/services/shipment_line_item/get_shipment_line_items.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+		}
+	} else if !session.IsOfficeUser() {
+		return nil, models.ErrFetchForbidden
+	}
+
+	shipmentLineItems, err := models.FetchLineItemsByShipmentID(i.db, &shipmentID)
+	if err != nil {
```
Comment (lynzt) [inline pkg/services/shipment_line_item/get_shipment_line_items.go]: do we want to return a reason for the error?

File: pkg/services/shipment_line_item/recalculate_shipment_line_items.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+type recalculateShipmentLineItems struct {
+	db     *pop.Connection
+	logger Logger
+}
+
+// RecalculateShipmentLineItems returns a collection of ShipmentLineItems that have been recalculated.
+func (r *recalculateShipmentLineItems) RecalculateShipmentLineItems(shipmentID uuid.UUID, session *auth.Session, route route.Planner) ([]models.ShipmentLineItem, error) {
+	if session.IsTspUser() {
+		// Check that the TSP user can access the shipment
+		_, _, err := models.FetchShipmentForVerifiedTSPUser(r.db, session.TspUserID, shipmentID)
```
Comment (lynzt) [inline pkg/services/shipment_line_item/recalculate_shipment_line_items.go]: reason for the errors in this file?

File: pkg/services/shipment_line_item/get_shipment_line_items_test.go [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+import (
+	"github.com/transcom/mymove/pkg/auth"
+	"github.com/transcom/mymove/pkg/models"
+	"github.com/transcom/mymove/pkg/testdatagen"
+)
+
+func (suite *ShipmentLineItemServiceSuite) TestGetShipmentLineItems() {
+	tspUser := testdatagen.MakeDefaultTspUser(suite.DB())
+	serviceMemberUser := testdatagen.MakeDefaultServiceMember(suite.DB())
+	//officeUser := testdatagen.MakeDefaultOfficeUser(suite.DB())
```
[the remark this answers is not in the document]
Comment (Ryan-Koch) [reply pkg/services/shipment_line_item/get_shipment_line_items_test.go]: Good catch, will do

File: pkg/services/shipment_line_item/get_shipment_line_items.go (hunk 1 above)
Comment (Ryan-Koch) [reply pkg/services/shipment_line_item/get_shipment_line_items.go]: The error object winds up passed along to the handler where it then winds up using the logger to document the failure. The error cause is passed up along from the originating function, in this case FetchShipmentForVerifiedTSPUser (which in most cases winds up being models.ErrFetchForbidden or models.ErrUserUnauthorized). The cause going along will allow the ResponseForError() used for handlers to do it's thing also.

**Pull request merged**